app/src/main/kotlin/com/wire/android/util/DateAndTimeParsers.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+@file:Suppress("TooGenericExceptionCaught")
+
+package com.wire.android.util
+
+import com.wire.android.appLogger
+import kotlinx.datetime.Instant
+import kotlinx.datetime.toJavaInstant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+import java.util.Date
+import java.util.Locale
+
+//region convenience ext functions, for retro compatibility
+fun String.deviceDateTimeFormat(): String? = DateAndTimeParsers.deviceDateTimeFormat(this)
+fun String.serverDate(): Date? = DateAndTimeParsers.serverDate(this)
+fun String.formatMediumDateTime(): String? = DateAndTimeParsers.formatMediumDateTime(this)
+fun String.formatFullDateShortTime(): String? = DateAndTimeParsers.formatFullDateShortTime(this)
+fun Instant.fileDateTime(): String = DateAndTimeParsers.fileDateTime(this)
+fun Instant.uiReadReceiptDateTime(): String = DateAndTimeParsers.uiReadReceiptDateTime(this)
+fun Date.toMediumOnlyDateTime(): String = DateAndTimeParsers.toMediumOnlyDateTime(this)
+fun String.uiMessageDateTime(): String? = DateAndTimeParsers.uiMessageDateTime(this)
```

Comment:
are those functions stable? meaning for the same input they always have the same output, if so we can annotate them with @Stable so they will not cause any unwanted recompression